web/v1/snowflake/sql-runner/sql/standard/00-setup/01-main/01-stored-procedures.sql

```diff
@@ -0,0 +1,426 @@
+  // == Helpers ==
+
+  function list_cols_with_type(sch,tbl,delimiter) {
+      var stmt = `
+          SELECT
+              LISTAGG(
+                  CASE
+                    WHEN isc.data_type='TEXT'
+                      THEN CONCAT(isc.column_name, ' VARCHAR(',isc.character_maximum_length, ')')
+                    WHEN isc.data_type='NUMBER'
+                      THEN CONCAT(isc.column_name, ' NUMBER(', isc.numeric_precision, ',',isc.numeric_scale, ')')
+                    ELSE
+                      CONCAT(isc.column_name, ' ', isc.data_type)
+                  END, '` + delimiter + `')
+              WITHIN GROUP (order by isc.ordinal_position)
+          FROM information_schema.columns AS isc
+          WHERE table_schema='` + sch + `'
+            AND table_name='` + tbl + `';`;
+
+      var res = snowflake.createStatement({sqlText: stmt}).execute();
+      res.next();
+      result = res.getColumnValue(1);
+
+      return result;
+  }
+
+  function notIncludedIn(arr) {
+      return function(elt) {
+        return ! arr.includes(elt);
+      };
+  }
+
+  function add_columns_to(sch, tbl, cols) {
+      var alter_stmt = `ALTER TABLE ` + sch + `.` + tbl + ` ADD COLUMN ` + cols;
+      snowflake.createStatement({sqlText: alter_stmt}).execute();
+
+      return "ok. Columns added.";
+  }
+
+  $$
+;
+
+
+/*
+  COMMIT_WITH_METADATA
+  Commits metadata and makes the necessary deletes/inserts for the commit steps in modules.
+  Wraps the above in a transaction.
+  Inputs:
+  META_TBL:                  the metadata table to commit (datamodel_metadata)
+  META_FROM_TBL:             the table from which to insert to metadata
+  STAGE_NXT:                 corresponds to '{{.stage_next}}'
+  STG_SRC_SCHEMA:            the schema for table as source
+  STG_SRC_TBL:               the source table name (to stage)
+  STG_TRG_SCHEMA:            the schema for table as target
+  STG_TRG_TBL:               the target table name (stage)
+  STG_JOIN_KEY:              the join key for staged tables
+  SKIP_DRV:                  corresponds to '{{.skip_derived}}'
+  DRV_SRC_SCHEMA:            the schema for table as source
+  DRV_SRC_TBL:               the source table name (to derive)
+  DRV_TRG_SCHEMA:            the schema for table as target
+  DRV_TRG_TBL:               the target table name (derive)
+  DRV_JOIN_KEY:              the join key for derived table
+  DRV_PARTITION_KEY:         the partition key for derived table
+  DRV_UPSERT_LOOKBACK FLOAT: corresponds to {{.upsert_lookback_days}}::float
+
+  To drop:
+  DROP PROCEDURE {{.output_schema}}.commit_with_metadata(VARCHAR,VARCHAR,VARCHAR,VARCHAR,
+                                                         VARCHAR,VARCHAR,VARCHAR,VARCHAR,
+                                                         VARCHAR,VARCHAR,VARCHAR,VARCHAR,
+                                                         VARCHAR,VARCHAR,VARCHAR,FLOAT);
+
+*/
+CREATE OR REPLACE PROCEDURE {{.output_schema}}.commit_with_metadata(
```

Comment:
I'm not sure I'm a fan of tying the commit to updating the metadata table - because for custom modules, I'm not sure we should force the customer to care about the metadata tables. 

Open to discussion.

Reply:
This is indeed something i have also been considering. The idea is that if someone wants metadata, it should be simple to accommodate in a custom module.
Would it make sense to add a flag as an argument, so that at least it is there for those who would prefer to do so?

Reply:
> The idea is that if someone wants metadata, it should be simple to accommodate in a custom module.

This I am on board with.

I think it's preferable to have an 'update_metadata' procedure as distinct from the commit procedure.

Reply:
Do you mean updating the metadata tables to go in its own transaction independent of the commit?
There is also the option to overload a commit procedure name, with different arguments, one with and one without committing metadata.